assets/app/features/upgrade/controllers/upgrade-nodes.controller.js

```diff
@@ -90,6 +138,8 @@
                             vm.nodesUpgrade.upgradedNodes = response.data.upgraded.length;
                             vm.nodesUpgrade.totalNodes =
                                 response.data.not_upgraded.length + vm.nodesUpgrade.upgradedNodes;
+                            upgradeStepsFactory.setUpgradeAll(vm.nodesUpgrade.computeUpgradeEnabled);
+                            upgradeStepsFactory.setUpgradeStep(NODE_UPGRADE_STEPS.controller);
```

Comment:
If this was supposed to initialize the step at the beginning of upgrade then it will not work again if you refresh after nodes upgrade is started (`remaining_nodes` will not be `null` anymore). Maybe set this based on the `current_substep` from status response? This piece of code was originally meant only to set the nodes counts before upgrade is started. Rest should be available from main status response.

Reply:
Will look into this.